### PR TITLE
Update nx and major dependencies

### PR DESCRIPTION
**Pull request by eirikurn**
# Update nx and major dependencies

## What

Upgrade to latest nx, next, nestjs, react, prettier, jest, eslint and typescript.

#RefactorWeek

## Why

To stay up to date with the latest features and bug fixes.

## Testing plan

- [x] Test 1 project of each type locally.
- [x] Test next build + docker locally.
- [x] Check if CI builds.
- [ ] Merge and deploy to dev.
- [ ] Ping every team to verify that their projects works on dev.

This is pretty manual. We need to improve our E2E testing story.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against master before asking for a review

### CODE REVIEW

**Comment by dabbeg** [comment]
When I start a web project (I tried `air-discount-scheme-web` and `adgerdir`) I get an error:

```
error - /home/dabbeg/repos/islandis/island.is/libs/island-ui/core/src/lib/Typography/Typography.treat.ts
TreatError: treat-webpack-plugin: An error occured during compilation: 
ReferenceError: $RefreshReg$ is not defined
```

Plus the tsconfig.json is automatically updated with some stuff

File: scripts/Dockerfile [hunk 1]
```diff
@@ -81,7 +81,7 @@ FROM output-base as output-next
 COPY --from=builder /build/${APP_HOME}/next.config.js /webapp
 COPY --from=builder /build/${APP_HOME}/next-modules/ /webapp/next-modules/
 COPY --from=builder /build/${APP_HOME}/public /webapp/public
-COPY --from=builder /build/${APP_DIST_HOME} /webapp/.next
+COPY --from=builder /build/${APP_DIST_HOME}/.next /webapp/.next
```
Comment (pshomov) [inline scripts/Dockerfile:84]: wtf, how did this work until now?!?

**Review by pshomov — DISMISSED** [review]
LGTM 🤞🏻

File: scripts/Dockerfile (hunk 1 above)
Comment (eirikurn) [reply scripts/Dockerfile:84]: It's a change in nx10, they now move a package.json, public and next.config.js to the dist folder, and move .next one level down.

**Comment by eirikurn** [comment]
@dabbeg is it possible you have an old version of this branch? I fixed these errors already.

**Review by dabbeg — APPROVED** [review]
My bad, I had an old version on my branch :facepalm:
Looks awesome!

**Review by barabrian — APPROVED** [review]
Good Job! 👏  👑

**Comment by eirikurn** [comment]
Finally! All checks have passed. Can I get fresh reviews from @barabrian, @dabbeg, @island-is/aranja, @island-is/sendiradid, @island-is/stefna and @island-is/kolibri?

Before there's a new conflict with master :)

**Review by barabrian — APPROVED** [review]
Love it!

**Review by gudjong — APPROVED** [review]
Nice

**Review by arnifreyrs — APPROVED** [review]

**Review by darrikonn — APPROVED** [review]
Nice

**Review by obmagnusson — APPROVED** [review]
LGTM

**Review by haukurh — APPROVED** [review]
LGTM

**Comment by eirikurn** [comment]
Thanks everyone!

**Pull request merged**